libs/ledger-live-common/src/hw/isDevFirmware.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * In order to mitigate users on odd firmware versions reporting
+ * connectivity errors when interacting with the app. Attempt to detect
+ * those versions and set a flag for UI.
+ */
+const isDevFirmware = (seVersion: string | undefined): boolean => {
+  if (!seVersion) return false;
+
+  const knownDevSuffixes = ["lo", "rc", "il", "tr"]; // FW can't guarantee non digits in versions
+  const pattern = new RegExp(`.+-(${knownDevSuffixes.join("|")})\\d{0,}`);
```

Comment:
that part is overkill i think.
is there a way NOT to use a regexp? RegExp construction may be slow in some cases.

would it work to do something like `seVersion.includes("-rc")` 🤔  ultimately, isn't the presence of a `-` a dev firmware? (maybe i'm wrong)

Reply:
No, @rgeslain-ledger said that's the only route. I don't think it's overkill though, we only run this when we run the `getDeviceInfo`, not on every frame or render.

Reply:
anyway, this should work

```js
const knownDevSuffixes = ["lo", "rc", "il", "tr"];
knownDevSuffixes.some(s => seVersion.includes("-"+s))
```

Reply:
Sure ... 🤷🏼